src/Stratis.StraxD/Program.cs

```diff
@@ -32,6 +36,8 @@ public static async Task Main(string[] args)
                     MinProtocolVersion = ProtocolVersion.PROVEN_HEADER_VERSION
                 };
 
+                await CheckLegacyGenesisHashAsync(nodeSettings);
```

Comment:
If you put this code in `UseNodeSettings` then it will work for any daemon. This relies on this particular daemon being run. What if another daemon is started up?

Reply:
There woudlnt have been any other daemon run ever ;)

Reply:
I would not be so sure that some other daemon couldn't be run later instead of this one. Not sure why you decided to make it specific to a daemon. There's really no advantage in that. But then again, I could be missing something.

Reply:
The majority would have ran v0.9 which would have had the incorrect genesis. If any other daemon was run they can be notified to clear their folder..